pkg/trait/util.go

```diff
@@ -320,3 +321,26 @@ func getBuilderTask(tasks []v1.Task) *v1.BuilderTask {
 	}
 	return nil
 }
+
+// Equals return if traits set using annotations have changed.
+func Equals(oldObject metav1.Object, newObject metav1.Object) bool {
```

Comment:
I first saw the function name and misunderstood it as comparing both annotations and spec.traits. It's only when I stared at the arg types that I understood the correct intent.

Apart from the function name, is it not needed to actually compare the traits under specs as well for where the func is invoked?  Are changes to traits under spec handled in a different mechanics?

Reply:
Yes I think the current solution is suboptimal but with my outdated knowledge of the codebase I tried to be conservative.

I think the proper solution would be to always reason on a synthetic view of the trait as in fact we may have cases where two resources have the same traits but defined in a different way.

I'll try to work on something

Reply:
pushed a  new version of the PR with better support for traits